platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/AjaxEngine.ts

```diff
@@ -0,0 +1,166 @@
+// class definition of DOMContext.
+class AjaxEngine {
+	private readonly nodes = new Map<number, Node>();
+	private zIndex = 100;
+
+	public allocateZIndex() {
+		this.zIndex += 1;
+		return '' + this.zIndex;
+	}
+
+	public setMaximumZIndex(node: HTMLElement) {
+		if(node.style.zIndex != '' + this.zIndex) {
+			node.style.zIndex = this.allocateZIndex();
+		}
+	}
+
+	public getNode(nodeId: number) {
+		return this.nodes.get(nodeId) as Node;
```

Comment:
`as Node` ... is this cast still necessary if we declare the function like
`public getNode(nodeId: number): Node {`
?

Reply:
yes, because, `get` returns `Node | null`, so we have to cast

Reply:
k